x-pack/plugins/fleet/server/services/agent_policy_create.ts

```diff
@@ -0,0 +1,146 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+
+import type { ElasticsearchClient, SavedObjectsClientContract } from 'src/core/server';
+
+import type { AuthenticatedUser } from '../../../security/common/model';
+
+import {
+  FLEET_ELASTIC_AGENT_PACKAGE,
+  FLEET_SERVER_PACKAGE,
+  FLEET_SYSTEM_PACKAGE,
+} from '../../common';
+
+import type { AgentPolicy, NewAgentPolicy } from '../types';
+
+import { agentPolicyService, packagePolicyService } from '.';
+import { incrementPackageName } from './package_policy';
+import { bulkInstallPackages } from './epm/packages';
+import { ensureDefaultEnrollmentAPIKeysExists } from './setup';
+
+const FLEET_SERVER_POLICY_ID = 'fleet-server-policy';
+
+async function getAgentPolicyId(soClient: SavedObjectsClientContract): Promise<string | undefined> {
```

Comment:
nit: could we improve the naming here `getFleetServerAgentPolicyId` maybe?  
Also do the id is important here? I think the agent working on docker only rely on the flag `is_default_fleet_server` this seems a little hacky to me and I am wondering if we documented well these decisions

Reply:
this was discussed in an earlier conversation on this pr and the id was suggested as a default after we remove the `is_default_fleet_server` flag: https://github.com/elastic/beats/issues/29774